lib/stud/task.rb

```diff
@@ -36,5 +36,14 @@ def stop!
     def self.interrupted?
       Thread.current[:stud_task_interrupted]
```

Comment:
Yes, there is a bug in the current implementation of stop! 
It should be the following code and not `Thread.current`, I believe `Thread.current` is OK in the context of `interrupted?` since its called inside the thread from stud interval.

```ruby
def stop!
  @thread[: stud_task_interrupted] = true
end
```